eventmesh/LockManager.js

```diff
@@ -118,29 +128,35 @@ class LockManager {
         }
       })
       .tap(() => logger.debug(`Successfully acquired lock on resource with resourceId: ${resourceId}`))
-      .catch(err => {
-        //TODO-PR - use catch filter
-        if (err instanceof NotFound) {
-          const spec = {
-            options: JSON.stringify(opts)
-          };
-          const status = {
-            locked: 'true'
-          };
-          const body = {
-            metadata: {
-              name: resourceId
-            },
-            spec: spec,
-            status: status
-          };
-          return eventmesh.apiServerClient.createLock(CONST.APISERVER.RESOURCE_TYPES.DEPLOYMENT_LOCKS, body)
-            .tap(() => logger.debug(`Successfully acquired lock on resource with resourceId: ${resourceId} `));
-        } else if (err instanceof Conflict) {
-          //TODO-PR - add details in DeploymentAlreadyLocked
-          throw new DeploymentAlreadyLocked(resourceId);
-        }
-        throw err;
+      .catch(NotFound, () => {
```

Comment:
Yes, for any other error, it will bubble up and middleware error handler will throw it. Problem with any other error is that we can not throw them as DeploymentAlreadyLocked as that will be wrong. Hence I guess better would be to let the middleware handle it.